### PR TITLE
Enrollment Wizard

### PR DESCRIPTION
**Pull request by TMazenge**
Added the enrollment wizard for Android. This is the initial UI for the wizard.

### CODE REVIEW

**Comment by eriyezab** [comment]
Regarding the conflicts, all the files that are in target/ directory should be deleted

File: src/main/webapp/enroll.html [hunk 1]
```diff
@@ -72,18 +103,31 @@ <h2>SDK Library Detail</h2>
             <div class="form-group">
             <label for="nv">Release Notes CL</label>
             <input type="text" class="form-control" id="rnc" name="release">
+
             </div>
             
-            <button type="submit" class="btn btn-primary">ENROLL</button>
+            <div class="panel panel-primary setup-content" id="step-4">
+                <div class="panel-heading">
+                    <h3 class="panel-title"> Link to Release CL Notes</h3>
```
Comment (shrutip90) [inline src/main/webapp/enroll.html]: Change "Release CL Notes" to "Release Notes CL"

File: src/main/webapp/enroll.html [hunk 2]
```diff
@@ -72,18 +103,31 @@ <h2>SDK Library Detail</h2>
...
+                </div>
+                <div class="panel-body">
+                    <div class="form-group">
+                        <label class="control-label">Release Notes</label>
+                        <input maxlength="200" type="text" required="required" class="form-control" placeholder="Enter Release CL Notes"/>
```
Comment (shrutip90) [inline src/main/webapp/enroll.html]: ditto

File: src/main/webapp/js/script.js [hunk 3]
```diff
@@ -34,3 +34,45 @@ function showOrHideNotes(version) {
...
+                isValid = false;
+                $(curInputs[i]).closest(".form-group").addClass("has-error");
+            }
+        }
+
+        if (isValid) nextStepWizard.removeAttr('disabled').trigger('click');
+    });
+
+    $('div.setup-panel div a.btn-success').trigger('click');
+});
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: Please take care of the warning.

File: src/main/webapp/release.html [hunk 4]
```diff
@@ -9,63 +9,116 @@
...
-                <div class="card">
-                    <h2 class="card-title">M78</h2>
+
+        <div class="col" >
+            <div class="row justify-content-center">
+                <div class="card text-white">
+                    <div class="card">
+                        <div class="card-header">M78</div>
+                          <div class="card-body">
+                              <p>Release Manager: ashwinraghav@</p>
```
Comment (shrutip90) [inline src/main/webapp/release.html]: Please remove this and add Enrolled SDK count instead.

File: src/main/webapp/release.html [hunk 5]
```diff
@@ -9,63 +9,116 @@
...
+                              <p>Cold Freeze August 18, 2020</p>
```
Comment (shrutip90) [inline src/main/webapp/release.html]: Add ":" after the name of the field.

File: src/main/webapp/enroll.html [hunk 6]
```diff
@@ -14,55 +20,80 @@
...
+                    <h3 class="panel-title">SDK</h3>
+                </div>
+                <div class="panel-body">
+                    <div class="form-group">
+                        <label class="control-label">Select SDK</label>
+                        <select name="sdks" class="custom-select mb-3">
+                        <option selected>Select SDK</option>
+                        <option value="firebase_common">Firebase Common</option>
+                        <option value="firebase_messaging">Firebase Messaging</option>
+                        <option value="firebase_idd">Firebase Idd</option>
```
Comment (shrutip90) [inline src/main/webapp/enroll.html]: -> IID

File: src/main/webapp/enroll.html [hunk 7]
```diff
@@ -14,55 +20,80 @@
...
+                    <a href="#step-2" type="button" class="btn btn-default btn-circle" disabled="disabled">2</a>
+                    <p><small>Current Version</small></p>
+                </div>
+                <div class="stepwizard-step col-xs-3"> 
+                    <a href="#step-3" type="button" class="btn btn-default btn-circle" disabled="disabled">3</a>
+                    <p><small>Version Being Released</small></p>
+                </div>
+                <div class="stepwizard-step col-xs-3"> 
+                    <a href="#step-4" type="button" class="btn btn-default btn-circle" disabled="disabled">4</a>
+                    <p><small>Link to Cl Notes</small></p>
```
Comment (shrutip90) [inline src/main/webapp/enroll.html]: I think the name of the enrollment step should be more generic than just repeating what input is being asked for in the step.
You can change this to "Release details", so we can add other fields later on the same page.

You could also avoid naming the steps completely. That might be easier and cleaner.

**Review by shrutip90 — CHANGES_REQUESTED** [review]

**Review by eriyezab — COMMENTED** [review]
nit: indentation should be 2 spaces not 4

**Comment by TMazenge** [comment]
> Regarding the conflicts, all the files that are in target/ directory should be deleted

Sure thing, will delete when I resolve the comments.

File: src/main/webapp/enroll.html (hunk 1 above)
Comment (TMazenge) [reply src/main/webapp/enroll.html]: Done.

File: src/main/webapp/enroll.html (hunk 2 above)
Comment (TMazenge) [reply src/main/webapp/enroll.html]: Done.

File: src/main/webapp/js/script.js (hunk 3 above)
Comment (TMazenge) [reply src/main/webapp/js/script.js]: Done.

File: src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.appengine.api.datastore.Query.SortDirection;
+import firebase.sdk.dashboard.data.User;
+import firebase.sdk.dashboard.data.Release;
+import firebase.sdk.dashboard.data.SDKReleaseMetadata;
+import firebase.sdk.dashboard.data.Platform;
+import firebase.sdk.dashboard.data.Platform;
+import java.util.List;
+
+/**
+ * An platform release dao implementation.
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java:17]: nit: Include details on what kind of implementation. You could say "Implementation of the Platform release DAO using Datastore for storage."

File: src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+
+/**
+ * An platform release dao implementation. 
+ */
+public class PlaformReleaseDaoDatastore implements PlaformReleaseDao {
+
+    DatastoreService platformDatastore = DatastoreServiceFactory.getDatastoreService();
+
+
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java:23]: nit: Remove extra blank line

File: src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java [hunk 10]
```diff
@@ -0,0 +1,53 @@
+package firebase.sdk.dashboard.dao;
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java:1]: I guess these changes are not supposed to be in this PR. Can you please move them to a separate branch/PR.

File: src/main/webapp/enroll.html [hunk 11]
```diff
@@ -14,55 +20,76 @@
...
             </div>
+            
+            <div class="panel panel-primary setup-content" id="step-3">
+                <div class="panel-heading">
+                    <h3 class="panel-title">Version Being Released</h3>
+                </div>
+                <div class="panel-body">
+                    <div class="form-group">
+                        <label class="control-label">Version Being Released</label>
+                        <input maxlength="200" type="text" required="required" class="form-control" placeholder="Enter Version being Released"/>
```
Comment (shrutip90) [inline src/main/webapp/enroll.html:85]: nit: -> 'version being released', since 'being' is not capitalized

File: src/main/webapp/index.html [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+      <title>SDK Dashboard</title>
+      
+      <link rel="stylesheet" href="css/landing.css">
+      <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M" crossorigin="anonymous">
+      <script src="https://kit.fontawesome.com/4588e35b97.js" crossorigin="anonymous"></script>    
+  </head>
+
+  <body>
+    <div class="jumbotron">
+        <h1>SDK Firebase Release Dashboard</h1>
```
Comment (shrutip90) [inline src/main/webapp/index.html:15]: nit: "Firebase SDK Release Dashboard"

File: src/main/webapp/release.html [hunk 13]
```diff
@@ -9,63 +9,117 @@
...
+
+        <div class="col" >
+            <div class="row justify-content-center">
+                <div class="card text-white">
+                    <div class="card">
+                        <div class="card-header">M78</div>
+                          <div class="card-body">
+                              <p>Enrolled SDKs: 85</p>
+                              <p>Cold Freeze: August 18, 2020</p>
+                              <p>Launch Deadline: August 21, 2020</p>
```
Comment (shrutip90) [inline src/main/webapp/release.html:26]: -> Launchcal Deadline. Please apply same change to the other lines in this file.

File: src/main/webapp/release_sdks.html [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+                <div class="sdk">
+                    Firebase Messaging       
+                </div>
+           </div>
+        </div>
+
+         <div class="col" >
+            <div class="row justify-content-center">
+                <div class="sdk">
+                    Firebase IDD
```
Comment (shrutip90) [inline src/main/webapp/release_sdks.html:36]: nit: IID

**Review by shrutip90 — APPROVED** [review]
Ignoring the changes in target/ files. Please let me know if I should take a look at those.

**Pull request merged**